server/src/main/java/com/cloud/servlet/ConsoleProxyServlet.java

```diff
@@ -500,6 +526,20 @@ private String composeConsoleAccessUrl(String rootUrl, VirtualMachine vm, HostVO
         return sb.toString();
     }
 
+    /**
+     * Generates an extra parameter to be passed to noVNC server to connect to WebSocket exposed on Vmware hosts.
+     * This method has been introduced as Vmware 7 has deprecated the previous VNC approach to connect to VMs console.
+     * Please check: https://docs.vmware.com/en/VMware-vSphere/7.0/rn/vsphere-esxi-vcenter-server-70-release-notes.html
+     */
+    private String generateExtraNoVNCParameter(ConsoleProxyClientParam param) {
+        String extraParam = null;
+        if (StringUtils.isNotBlank(param.getTicketAcquired())) {
+            extraParam = "wss://" + param.getClientHostAddress() + ":443/ticket/" + param.getTicketAcquired();
```

Comment:
@nvazquez declare a constant for port 443 in some common class (may be ConsoleProxyManager), and use it across.